elements/howto-toggle-button/howto-toggle-button.js

```diff
@@ -0,0 +1,233 @@
+/**
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+(function() {
+  /**
+   * Define key codes to help with handling keyboard events.
+   */
+  const KEYCODE = {
+    SPACE: 32,
+    ENTER: 13,
+  };
+
+  /**
+   * Cloning contents from a &lt;template&gt; element is more performant
+   * than using innerHTML because it avoids addtional HTML parse costs.
+   */
+  const template = document.createElement('template');
+  template.innerHTML = `
+    <style>
+      :host {
+        background-color: #eee;
```

Comment:
We try to keep visual styling out of the element. I’d say move stuff like `padding`, `background-color`, `border` etc to the demo file

Reply:
I was a little confused about this one, I understand why. But, I saw that the howto-checkbox was doing it's styling through images and I needed similar basic styles for the button to work out of the box, I'll move them to the demo page instead. @robdodson any suggestions here?

Reply:
I used images because they were sort of the easiest path to drawing a checkmark. But I've been planning to change that to something less opinionated. In general, we try to only use styles for the very basics of rendering the element correctly. This is partly to discourage folks from thinking this is a for-production element set :)